#!/usr/bin/env python
PACKAGE = "bitbots_vision_tools"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("debug_on",   bool_t,   0, "Turns debug-mode on",  None)

gen.add("debug_printer_classes", str_t, 0, "debug_printer_classes", None)

gen.add("use_pixel_list",   bool_t,   0, "Use_Pixellist_TRUE_HsvSpace_FALSE",  None)
gen.add("pixel_list_path",    str_t,    0, "packagePath",  "/config/testfield_color01.yaml", None)

gen.add("h_min",    int_t,    0, "h_min", 0,  0, 255)
gen.add("s_min",    int_t,    0, "s_min", 0,  0, 255)
gen.add("v_min",    int_t,    0, "v_min", 180,  0, 255)
gen.add("h_max",    int_t,    0, "h_max", 255,  0, 255)
gen.add("s_max",    int_t,    0, "s_max", 60,  0, 255)
gen.add("v_max",    int_t,    0, "v_max", 255,  0, 255)


exit(gen.generate(PACKAGE, "color_test", "ColorTest"))
